src/makeInfinityAdder.js

```diff
@@ -6,7 +6,44 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let currentSum = 0;
+  let count = 0;
+  const adder = (number) => {
+    if (number === undefined && count === 0) {
+      count++;
+
+      return currentSum;
+    }
+
+    if (number === undefined && count === 1) {
+      currentSum = 0;
+
+      return currentSum;
+    }
+
+    currentSum = number;
+
+    const result = (num) => {
+      if (num === undefined && count === 0) {
+        count++;
+
+        return currentSum;
+      }
+
+      if (num === undefined && count === 1) {
+        currentSum = 0;
+
+        return currentSum;
+      }
+      currentSum += num;
+
+      return result;
+    };
```

Comment:
avoid code duplicate. Think about optimize. If you need you can use helper functions or common constants